compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -669,11 +669,14 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
         )
         .fold_with(&mut collector);
 
-    debug_assert_ne!(
-        collector.types.len(),
-        0,
-        "expect >1 RPITITs in call to `collect_return_position_impl_trait_in_trait_tys`"
-    );
+    if let ty::Error(_) = unnormalized_trait_sig.output().kind() {
```

Comment:
You need to check if this [`references_error`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/visit/trait.TypeVisitableExt.html#method.references_error), otherwise I'm pretty sure that the given code continues to fail because there are ways to get return types that have embedded `ty::Error`:

```rust
#![feature(return_position_impl_trait_in_trait)]

struct Wrapper<T>(T);

trait Foo {
    fn bar() -> Wrapper<Missing<impl Sized>>;
    //~^ ERROR: cannot find type `Missing` in this scope [E0412]
}

impl Foo for () {
    fn bar() -> Wrapper<i32> {
        Wrapper(0)
    }
}

fn main() {}
```

Reply:
Actually maybe is better to add this as the test case.